### ISSUE TITLE
Allow to run the test cases without interaction (Was: 4.93.2: test suite fails)

### ISSUE THREAD

**Issue by kloczek** [lsof-org/lsof#61]
<pre>+ /usr/bin/make -O -j48 -C tests test
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` -c LTlib.c \
-o LTlib.o
LTlib.c: In function &apos;ConvLsofDev&apos;:
LTlib.c:246:25: warning: implicit declaration of function &apos;major&apos;; did you mean &apos;major_X&apos;? [-Wimplicit-function-declaration]
  246 | #define major_X(dp, em) major(x2dev(dp, em))
      |                         ^~~~~
LTlib.c:375:31: note: in expansion of macro &apos;major_X&apos;
  375 |     ldev-&gt;maj = (unsigned int)major_X(dp, &amp;em);
      |                               ^~~~~~~
LTlib.c:250:25: warning: implicit declaration of function &apos;minor&apos; [-Wimplicit-function-declaration]
  250 | #define minor_X(dp, em) minor(x2dev(dp, em))
      |                         ^~~~~
LTlib.c:378:31: note: in expansion of macro &apos;minor_X&apos;
  378 |     ldev-&gt;min = (unsigned int)minor_X(dp, &amp;em);
      |                               ^~~~~~~
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTbasic.c \
LTlib.o `cat config.xobj` -o LTbasic
/usr/bin/ld: LTlib.o: in function `ConvLsofDev&apos;:
LTlib.c:(.text+0xf8): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x12e): undefined reference to `minor&apos;
/usr/bin/ld: LTlib.o: in function `ConvStatDev&apos;:
LTlib.c:(.text+0x197): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x1b3): undefined reference to `minor&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:74: LTbasic] Error 1
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: *** Waiting for unfinished jobs....
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTnlink.c \
LTlib.o `cat config.xobj` -o LTnlink
/usr/bin/ld: LTlib.o: in function `ConvLsofDev&apos;:
LTlib.c:(.text+0xf8): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x12e): undefined reference to `minor&apos;
/usr/bin/ld: LTlib.o: in function `ConvStatDev&apos;:
LTlib.c:(.text+0x197): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x1b3): undefined reference to `minor&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:98: LTnlink] Error 1
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTszoff.c \
LTlib.o `cat config.xobj` -o LTszoff
/usr/bin/ld: LTlib.o: in function `ConvLsofDev&apos;:
LTlib.c:(.text+0xf8): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x12e): undefined reference to `minor&apos;
/usr/bin/ld: LTlib.o: in function `ConvStatDev&apos;:
LTlib.c:(.text+0x197): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x1b3): undefined reference to `minor&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:106: LTszoff] Error 1
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTunix.c \
LTlib.o `cat config.xobj` -o LTunix `cat config.ldflags`
/usr/bin/ld: LTlib.o: in function `ConvLsofDev&apos;:
LTlib.c:(.text+0xf8): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x12e): undefined reference to `minor&apos;
/usr/bin/ld: LTlib.o: in function `ConvStatDev&apos;:
LTlib.c:(.text+0x197): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x1b3): undefined reference to `minor&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:110: LTunix] Error 1
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTsock.c \
LTlib.o `cat config.xobj` -o LTsock `cat config.ldflags`
/usr/bin/ld: LTlib.o: in function `ConvLsofDev&apos;:
LTlib.c:(.text+0xf8): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x12e): undefined reference to `minor&apos;
/usr/bin/ld: LTlib.o: in function `ConvStatDev&apos;:
LTlib.c:(.text+0x197): undefined reference to `major&apos;
/usr/bin/ld: LTlib.c:(.text+0x1b3): undefined reference to `minor&apos;
collect2: error: ld returned 1 exit status
</pre>

**Comment by masatake** [comment]
Please, show the spec file.

**Comment by kloczek** [comment]
Yeah .. that is the problem that lsof has no proper ac/am/lt or meson or cmake build framework :)

**Comment by kloczek** [comment]
I don't think that it will help you but please find my spec file.

<pre>%global		build_cflags	%{build_cflags} -Os

Summary:	A utility which lists open files on a Linux/UNIX system
Name:		lsof
Version:	4.93.2
Release:	2%{?dist}
# Sendmail .. lib/snpf.c
# LGPLv2+  .. lib/regex.c, regex.h
License:	zlib and Sendmail and LGPLv2+
URL:		https://people.freebsd.org/~abe/
VCS:		https://github.com/lsof-org/lsof/
Source0:	%{VCS}/archive/%{version}.tar.gz#/%{name}-%{version}.tar.gz
Source1:	upstream2downstream.sh
Patch0:		%{name}-man-page-section.patch
BuildRequires:	gcc
BuildRequires:	groff-base
BuildRequires:	libselinux-devel
BuildRequires:	libtirpc-devel

%description
Lsof stands for LiSt Open Files, and it does just that: it lists information
about files that are open by the processes running on a UNIX system.

%prep
%autosetup -p1

%build
%{set_build_flags} \
./Configure -n linux
%{set_build_flags} \
%make_build DEBUG=&quot;%{build_cflags} -I%{_includedir}/tirpc&quot; CFGL=&quot;%{build_ldflags} -L./lib -llsof -lselinux -ltirpc&quot;
soelim -r Lsof.8 &gt; lsof.1

%check
%{set_build_flags} \
#make_build -C tests test

%install
%__install -m755 lsof -Dt %{buildroot}%{_bindir}
%__install lsof.1 -D %{buildroot}%{_mandir}/man1/lsof.1

%files
%doc 00FAQ 00LSOF-L 00QUICKSTART
%attr(755,root,root) %{_bindir}/lsof
%{_mandir}/man*/*
</pre>

**Comment by masatake** [comment]
I found I fixed in https://github.com/lsof-org/lsof/commit/d290aaa952e24150cf96bb73a517567480b15e4d .

**Comment by kloczek** [comment]
Your patch fixed compiling test suit (thanks for that) but I just found that this process does not provide batch mode because it ask to answer on interactive question.

Question: would you welcome some other build suite am/c/lt based?

Other thing. I just found that opt test fails but it does not exits with non-zero exit code so my package build finished successfully.

<pre>+ /usr/bin/make -O -j 48 -C tests test
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` -c LTlib.c \
-o LTlib.o
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTbasic.c \
LTlib.o `cat config.xobj` -o LTbasic
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTszoff.c \
LTlib.o `cat config.xobj` -o LTszoff
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTunix.c \
LTlib.o `cat config.xobj` -o LTunix `cat config.ldflags`
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTnlink.c \
LTlib.o `cat config.xobj` -o LTnlink
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTsock.c \
LTlib.o `cat config.xobj` -o LTsock `cat config.ldflags`
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;

==================================================================

!!!WARNING!!!

This dialect or its particular version may not have been validated
with the lsof test suite.  Consequently some tests may fail or may
not even compile.

This is the computed identity of this dialect, not found in the
test data base file, ./TestDB:

    LT_BIGF LT_CC LT_DIAL_linux LT_VERS=52000 _FILE_OFFSET_BITS=64

Do you want to continue (y|n) [n]? y
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;

Basic test:
LTbasic ... OK

Standard tests:
LTnlink ... OK
LTsock ... OK
LTszoff ... OK
LTunix ... OK
All standard tests succeeded.

Suggestion: try the optional tests: &quot;make opt&quot;

make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
+ export &apos;CFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto -Os&apos;
+ CFLAGS=&apos;-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto -Os&apos;
+ export &apos;CXXFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto&apos;
+ CXXFLAGS=&apos;-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto&apos;
+ export &apos;FFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules -flto&apos;
+ FFLAGS=&apos;-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules -flto&apos;
+ export &apos;FCFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules -flto&apos;
+ FCFLAGS=&apos;-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules -flto&apos;
+ export &apos;LDFLAGS=-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin&apos;
+ LDFLAGS=&apos;-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin&apos;
+ export AR=gcc-ar RANLIB=gcc-ranlib NM=gcc-nm
+ AR=gcc-ar
+ RANLIB=gcc-ranlib
+ NM=gcc-nm
+ export CC=gcc CXX=g++
+ CC=gcc
+ CXX=g++
+ /usr/bin/make -O -j 48 -C tests opt
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTbigf.c \
LTlib.o `cat config.xobj` -o LTbigf
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTdnlc.c \
LTlib.o `cat config.xobj` -o LTdnlc
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTlock.c \
LTlib.o `cat config.xobj` -o LTlock
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
`cat config.cc`  -I. -I.. `cat config.cflags` LTnfs.c \
LTlib.o `cat config.xobj` -o LTnfs
make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
make: Entering directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;

Optional tests:
LTbigf ... OK
LTdnlc ... /home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests found: 100.00%
           OK
LTlock ... lock mismatch: expected W, got &quot; &quot;
           lock mismatch: expected R, got &quot; &quot;
           lock mismatch: expected w, got &quot; &quot;
           lock mismatch: expected r, got &quot; &quot;
LTnfs ... ERROR!!!  test file ./config.LTnfs33859 not found by lsof
          Hint: the temporary path generated by this test might
          not be on an NFS file system, or this test might be
          unable to create a file on the NFS file system.

          As a work-around use the -p option to specify a path to
          a regular file (not a directory) on an NFS file system
          to which this test will have read access.

          See 00FAQ and 00TEST for more information.
Failed tests: 2

make: Leaving directory &apos;/home/tkloczko/rpmbuild/BUILD/lsof-4.93.2/tests&apos;
+ exit 0
Processing files: lsof-4.93.2-2.fc31.x86_64
Executing(%doc): /bin/sh -e /var/tmp/rpm-tmp.vNVukW
+ umask 022
+ cd /home/tkloczko/rpmbuild/BUILD
+ cd lsof-4.93.2
+ DOCDIR=/home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64/usr/share/doc/lsof
+ export LC_ALL=C
+ LC_ALL=C
+ export DOCDIR
+ /usr/bin/mkdir -p /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64/usr/share/doc/lsof
+ cp -pr 00FAQ /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64/usr/share/doc/lsof
+ cp -pr 00LSOF-L /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64/usr/share/doc/lsof
+ cp -pr 00QUICKSTART /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64/usr/share/doc/lsof
+ exit 0
Provides: lsof = 4.93.2-2.fc31 lsof(x86-64) = 4.93.2-2.fc31
Requires(rpmlib): rpmlib(CompressedFileNames) &lt;= 3.0.4-1 rpmlib(FileDigests) &lt;= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) &lt;= 4.0-1
Requires: libc.so.6()(64bit) libc.so.6(GLIBC_2.11)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.3)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libselinux.so.1()(64bit) libtirpc.so.3()(64bit) libtirpc.so.3(TIRPC_0.3.0)(64bit) rtld(GNU_HASH)
Processing files: lsof-debugsource-4.93.2-2.fc31.x86_64
Provides: lsof-debugsource = 4.93.2-2.fc31 lsof-debugsource(x86-64) = 4.93.2-2.fc31
Requires(rpmlib): rpmlib(CompressedFileNames) &lt;= 3.0.4-1 rpmlib(FileDigests) &lt;= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) &lt;= 4.0-1
Processing files: lsof-debuginfo-4.93.2-2.fc31.x86_64
Provides: debuginfo(build-id) = d9bfac459976e925e66503f4c7e8e7e3377268ad lsof-debuginfo = 4.93.2-2.fc31 lsof-debuginfo(x86-64) = 4.93.2-2.fc31
Requires(rpmlib): rpmlib(CompressedFileNames) &lt;= 3.0.4-1 rpmlib(FileDigests) &lt;= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) &lt;= 4.0-1
Recommends: lsof-debugsource(x86-64) = 4.93.2-2.fc31
Checking for unpackaged file(s): /usr/lib/rpm/check-files /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64
Wrote: /home/tkloczko/rpmbuild/SRPMS/lsof-4.93.2-2.fc31.src.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/lsof-4.93.2-2.fc31.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/lsof-debugsource-4.93.2-2.fc31.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/lsof-debuginfo-4.93.2-2.fc31.x86_64.rpm
Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.rRdMFX
+ umask 022
+ cd /home/tkloczko/rpmbuild/BUILD
+ cd lsof-4.93.2
+ /usr/bin/rm -rf /home/tkloczko/rpmbuild/BUILDROOT/lsof-4.93.2-2.fc31.x86_64
+ exit 0
</pre>

**Comment by masatake** [comment]
> Question: would you welcome some other build suite am/c/lt based?

I myself have no plan to do it.
I will accept pull requests for introducing autotools as far as the change doesn't break building lsof on FreeBSD. However, in that case, I think the person (may be you) who made the change should take over this project.


What our team use is check.bash.
We run "make opt" in tests/case-14-classic-opt.bash run from the check.bash.

To make LTnfs test run successfully, we may have to provide nfs directory. 
14-classic-opt.bash should do the preparation. It is at the bottom of my TODO list.

I wonder why LTlock is failed on your platform.

About LTnfs, I have to use the phrase, a pull request is welcome.
About LTlock, I think I have to inspect.  
Are you working on Fedora? If the answer is yes, I raise the priority.

**Comment by subnut** [comment]
@dilinger If you really want to, you can simply remove the `./tests/CkTestDB` file and create a symlink to `/usr/bin/true`, bypassing platform validation and ensuring tests are non-interactive.

**Comment by dilinger** [comment]
> @dilinger @kloczek If you really want to, you can simply remove the `./tests/CkTestDB` file and create a symlink to `/usr/bin/true`, bypassing platform validation and ensuring tests are non-interactive.

Thanks, but that's not really what I want. At the moment we (Debian) run the tests except for the NFS test. Debian in particular builds for at least 9 architectures, and we have a bunch of different (non-release, lower priority and sometimes really weird) architectures that we also build for like sparc64 and kfreebsd-i386, for a total of 22 architectures. In general, extra tests are valuable; but only if they make the build itself fail upon a failed test.

The database itself is also very strange.  Tests are skipped if run on a distribution with a 5.0.* Linux kernel (like Fedora 30, which was EOL'd 2 years ago), and some really ancient kernels like 2.4.*, but otherwise they're going to always run on any Linux distribution released in the past 3 years.

Sp either the tests are valuable for checking whether newer Linux kernels (and of course random other OSes) behave properly, in which case they should actually cause a build to fail if they hit a problem, or they're not valuable because they check for features that any modern *nix variant has had for decades. If the latter case, if they're not useful for any OS releases from the past decade, we should probably either delete the tests or make the platform check smarter to only run the tests on old OSes/kernels. Or skip the validated platform check altogether, and only run the tests when `make check` is run.

We might want to skip the validated platform check altogether, as at the moment I'm betting like 99% of lsof builds happen on non-validated platforms.

I do also think the NFS test needs to be fixed, which is what this bug was actually about, but I also wanted to take a step back and consider whether or not we even need those tests and that validated platform db.

**Comment by kloczek** [comment]
> @dilinger @kloczek If you really want to, you can simply remove the `./tests/CkTestDB` file and create a symlink to `/usr/bin/true`, bypassing platform validation and ensuring tests are non-interactive.

None of the test suites in thousands OSS projects are interractive.

**Comment by jiegec** [comment]
Please try the new autotools build system and report bugs if any.
